client/src/components/BlockEditor/BlockEditorEditing.tsx

```diff
@@ -201,7 +240,31 @@ export class BlockEditorEditingPresentational extends React.Component<
     }
   };
 
-  private onChangeCallback = (c: any) => {
+  private onChangeCallback = (c: Change) => {
+    // first check to see if document changed, which we can do with !== b/c
+    // Slate uses immutable objects, if it has changed then normalize wrt
+    // pointer spacing
+
+    if (c.value.document !== this.props.block.value.document) {
+      c = normalizeExportSpacing(c);
+    }
+
+    // the rest of this method before the last two lines are trying to prevent
+    // the user from using a click top get the text cursor in off limit spacing
+    // the onClick handler didn't seem to have access to the new cursor position
+    // so this logic lives here instead
+
+    const value = c.value;
+    const selection = value.selection;
+    const selectionIsExpanded = selection.isExpanded;
+
+    // if selection is expanded, then we don't do this
+    // this allows mouse dragging to select across pointers
+    if (!selectionIsExpanded && isCursorInPotentiallyProblematicPosition(value)) {
+      const correctedChange = handleStationaryCursor(c);
+      this.props.updateBlock({ id: this.props.block.id, value: correctedChange.value, pointerChanged: false });
```

Comment:
The first thing that onChange() does is call ``updateBlock``. So in this case updateBlock would be called twice. Is there any way we could compress this? Could the value in line 268 just be the ``correctedChange.value`` in this case?